File: CPP/sorting/insertionsort.cpp
#include"bits/stdc++.h"
using namespace std;
void insertion_sort(int arr[],int n){
    for (int i = 1; i < n; i++) {
        int temp=arr[i];
        int j=i-1;//we are using j here because we are changing j+1 th elemnt to the temp
        //ulta looop chalake check krenge aur shift krnge(if the element is greater than temp or not
        
        for (;j>=0;j--) {
            if(arr[j]>temp)
                arr[j+1]=arr[j];
            else
                break;
        }
        arr[j+1]=temp;
    }
}
int main(){
    int arr[]={5,3,2,4,1};
    int n=sizeof(arr)/sizeof(int);
    insertion_sort(arr,n);
    for(auto it: arr)
    cout<<it<<", ";
}

File: Java/SortingTechniques/InsertionSort.java
public class InsertionSort {
    public static void main(String[] args) {
        int [] arr = {2,6,1,3,8,5};
        for(int i=1;i<arr.length;i++){
            int tp = arr[i];
            int j=i-1;
            while(j>=0 && arr[j]>tp){
                arr[j+1]=arr[j];
                j--;
            }
            arr[j+1]=tp;
        }
        for(int no : arr){
            System.out.println(no);
        }
    }
}
